src/controllers/usersController.js

```diff
@@ -0,0 +1,124 @@
+import bcrypt from 'bcrypt';
+import { Op } from 'sequelize';
+import jwt from 'jsonwebtoken';
+import successRes from '../helpers/successHandler';
+import errorRes from '../helpers/errorHandler';
+import Models from '../database/models';
+import generatePassword from '../utils/passwordGenerator';
+import sendEmail from '../utils/mail2';
+import hashPwd from '../helpers/pwd';
+
+const { User, Role } = Models;
+
+export const register = async (req, res) => {
+  const validRole = await Role.findOne({ where: { role: req.body.role } });
+  if (!validRole) errorRes(res, 404, `Role ${req.body.role} is not allowed`);
+
+  try {
+    const userFromToken = req.user;
+    const signedUser = await User.findOne({ where: { id: userFromToken.id } });
+
+    if (signedUser.role === 'operator') {
+      if (req.body.role === 'operator') {
+        return errorRes(res, 401, 'Please sign in as admin');
+      }
+    }
+
+    const generatedPwd = generatePassword();
+    const hash = await hashPwd(generatedPwd);
+    const user = await User.create({
+      ...req.body,
+      password: hash,
+      verficationLink: '',
+      comfirmed: false,
+      resetLink: '',
+    });
+    const verficationLink = `http://localhost:${process.env.PORT}/api/users/verify/${user.id}`;
+    const resetLink = `http://localhost:${process.env.PORT}/api/users/reset/${user.id}`;
```

Comment:
same as here. put the value of restLink in the env variable

Reply:
Done also